File: CPE1星/UVA_10093.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    string str;

    while(cin>>str){

        int max=0;
        bool flag = true;
        for(int i=0;i<str.length();i++){
            if(('0'<=str[i] && str[i]<='9') || ('A'<=str[i] && str[i]<='Z') || ('a'<=str[i] && str[i]<='z') ){
                if(str[i]-'0'>max) max = str[i]-'0';
            }else{
                flag = false;
                break;  
            }
        }

        if(max<1){
            cout<<"such number is impossible!"<<endl; 
            continue;
        }

        if(max>=49){ //a-z
            max -= 6; 
        }if(max>9){ //A-Z
            max -= 7;
        }

        if(flag){
            cout<<max + 1<<endl;
        }else{
            cout<<"such number is impossible!"<<endl; 
        }     
        
    }
}

File: python/cpe_one_star/str/UVA-10093.py
while True:
    try:
        s = input()
        base = 1
        sum = 0
        for i in s:
            temp = 0
            if i.isalnum():
                if i.isnumeric():
                    base = max(base, int(i))
                    temp = int(i)
                elif i.isupper():
                    base = max(base, ord(i) - ord('A') + 10)
                    temp = ord(i) - ord('A') + 10
                elif i.islower():
                    base = max(base, ord(i) - ord('a') + 10 + 26)
                    temp = ord(i) - ord('a') + 10 + 26
            else:
                continue
            sum += temp
        for ans in range(base,62):
            if sum % ans == 0:
                print(ans + 1)
                break
        else:
            print("such number is impossible!")
    except:
        break